lib/commands/autocomplete/helper.js

```diff
@@ -0,0 +1,61 @@
+const fs = require('fs-extra');
+const path = require('path');
+
+module.exports = class Helper {
+    constructor(omelette, commanders = []) {
+        this.commanders = commanders;
+        this.completion = omelette('ask');
+        this.autoCompleteHintsFile = path.join(__dirname, 'autocomplete-hints.json');
+    }
+
+    _getAutoCompleteOptions() {
+        const options = {};
+        this.commanders.forEach(com => {
+            options[com.name()] = com.commands.map(sumCom => sumCom.name());
+        });
+
+        return options;
+    }
+
+    /**
+     * Initializes auto complete inside of the program
+     */
+    initAutoComplete() {
+        if (fs.existsSync(this.autoCompleteHintsFile)) {
+            const options = fs.readJsonSync(this.autoCompleteHintsFile);
+
+            this.completion.tree(options);
+            this.completion.init();
+        }
+    }
+
+    _withProcessExitDisabled(fn) {
+        const origExit = process.exit;
+        process.exit = () => {};
+        fn();
+        process.exit = origExit;
+    }
+
+    /**
+     * Regenerates auto complete hints file
+     */
+    reloadAutoCompleteHints() {
+        const options = this._getAutoCompleteOptions();
+        fs.writeJSONSync(this.autoCompleteHintsFile, options);
+    }
+
+    /**
+     * Sets ups auto complete. For example, adds autocomplete entry to .bash_profile file
+     */
+    setUpAutoComplete() {
+        this.reloadAutoCompleteHints();
+        this._withProcessExitDisabled(() => this.completion.setupShellInitFile());
```

Comment:
II think we need a try catch to cover this setupShellInitFile function, as suggested by the user https://github.com/f/omelette#automated-install. This step has too many potential problems depending on user's machine.

We need to do the cleanup if it fails, and pop up the message for the failure

Reply:
What type of clean up? How can we know what needs to be cleaned up? If this steps fails it will show user the error message from omelette. What extra can we do in try and catch except of rethrowing the error?

Reply:
When i run in windows here is the result:
```
PS D:\Github\ask-cli> ask autocomplete setup
D:\Github\ask-cli\node_modules\omelette\src\omelette.js:189
          throw new Error('Shell could not be detected');
          ^

Error: Shell could not be detected
    at Omelette.getActiveShell (D:\Github\ask-cli\node_modules\omelette\src\omelette.js:189:17)
    at Omelette.getDefaultShellInitFile (D:\Github\ask-cli\node_modules\omelette\src\omelette.js:210:35)
    at Omelette.setupShellInitFile (D:\Github\ask-cli\node_modules\omelette\src\omelette.js:238:42)
    at D:\Github\ask-cli\lib\commands\autocomplete\helper.js:52:61
    at Helper._withProcessExitDisabled (D:\Github\ask-cli\lib\commands\autocomplete\helper.js:35:9)
    at Helper.setUpAutoComplete (D:\Github\ask-cli\lib\commands\autocomplete\helper.js:52:14)
    at Command.<anonymous> (D:\Github\ask-cli\lib\commands\autocomplete\index.js:29:20)
    at Command.listener (D:\Github\ask-cli\node_modules\commander\index.js:370:29)
    at Command.emit (events.js:314:20)
    at Command.EventEmitter.emit (domain.js:486:12)
```

Should we pre-check the OS? And give a good message if it falls through?